### PR TITLE
Cache physical values only in buffer in ReducedSpaceEvaluator

### PR DESCRIPTION
**Pull request by frapac**
Before this PR, when using a `ReducedSpaceEvaluator` object `nlp`, values were cached both in `nlp.buffer` and in a state `nlp.x`. This introduced confusion and redundancy. This PR refactors the object `ReducedSpaceEvaluator` to keep only `nlp.buffer` as a cache. 

Additional details about this PR:
- **Warning**: it introduces a breaking change in the constructor of `ReducedSpaceEvaluator`, as we do not need anymore to pass an initial state to instantiate the evaluator object. 
- it improves the interface of the cache object `PolarNetworkState`, to ensure that `pinj` is always equal to `Cg * Pg - Pd`, and equivalently for `qinj`. That was not the case before. 
- hence, it should fix a few bugs we observed in  the ProxAL/ExaPF wrapper, and the convergence of the Newton-Raphson algorithm should be improved.

### CODE REVIEW

File: src/Evaluators/reduced_evaluator.jl [hunk 1]
```diff
@@ -344,9 +360,7 @@ end
 function reset!(nlp::ReducedSpaceEvaluator)
     # Reset adjoint
     fill!(nlp.λ, 0)
-    # Reset initial state
-    copy!(nlp.x, initial(nlp.model, State()))
     # Reset buffer
-    nlp.buffer = get(nlp.model, PhysicalState())
+    init!(nlp.model, nlp.buffer)
```
Comment (michel2323) [inline src/Evaluators/reduced_evaluator.jl:364]: This is just a general comment. I like the mutliple dispatch in Julia, but I think it makes the function names at times a bit too generic. We have `get`, `init!`, `transfer`, etc, and the function signature has to tell what gets transferred to what, which can be ambiguous. `init_buffer` for example would be more readable to me.

**Review by michel2323 — APPROVED** [review]

File: src/Evaluators/reduced_evaluator.jl [hunk 2]
```diff
@@ -131,10 +157,9 @@ bounds(nlp::ReducedSpaceEvaluator, ::Variables) = (nlp.u_min, nlp.u_max)
 bounds(nlp::ReducedSpaceEvaluator, ::Constraints) = (nlp.g_min, nlp.g_max)
 
 function update!(nlp::ReducedSpaceEvaluator, u; verbose_level=0)
-    x₀ = nlp.x
     jac_x = nlp.autodiff.Jgₓ
     # Transfer x, u into the network cache
```
[the remark this answers is not in the document]
Comment (frapac) [reply src/Evaluators/reduced_evaluator.jl:161]: thanks! I should have updated the comments

File: src/Evaluators/reduced_evaluator.jl (hunk 1 above)
Comment (frapac) [reply src/Evaluators/reduced_evaluator.jl:364]: you are right about that. I will change the name of the function

File: src/Evaluators/reduced_evaluator.jl [hunk 3]
```diff
@@ -149,9 +174,7 @@ function update!(nlp::ReducedSpaceEvaluator, u; verbose_level=0)
         LinearSolvers.update!(nlp.linear_solver, nlp.∇gᵗ)
     end
 
-    # Update value of nlp.x with new network state
-    get!(nlp.model, State(), nlp.x, nlp.buffer)
-    # Refresh value of the active power of the generators
+    # Refresh values of active and reactive powers at generators
     refresh!(nlp.model, PS.Generator(), PS.ActivePower(), nlp.buffer)
```
[the remark this answers is not in the document]
Comment (frapac) [reply src/Evaluators/reduced_evaluator.jl:178]: that's a good idea

**Pull request merged**